### DISCUSSION TITLE
Add a leximin optimization objective

### DISCUSSION THREAD

**Discussion by erelsgl** [cvxpy/cvxpy#1505] [Ideas]
*Leximin optimization* is a criterion for multi-objective optimization. It means that we first maximize the smallest objective; subject to this, we maximize the next-smallest objective; and so on.  See [Leximin order](https://en.wikipedia.org/wiki/Leximin_order) for more information and applications.

I have some code that implements leximin optimization for convex problems, and I will be happy to add it to cvxpy. Are you interested?

**Comment by rileyjmurray** [comment]
Hi @erelsgl! That sounds pretty interesting. Could you share the code you have so far?

**Comment by erelsgl** [comment]
Here:

https://github.com/erelsgl/fairpy/blob/master/items/leximin_generic.py

**Comment by erelsgl** [comment]
Would you like me to make a pull request? Where exactly can I make the change?

**Reply by akshayka** [reply]
Integrating this into CVXPY proper might require a fair amount of work. But it would still be great to get this into the CVXPY ecosystem somehow.

Have you thought about releasing this code as a small PyPi package, that has CVXPY as a dependency? If you do that, we could highlight it here: https://www.cvxpy.org/related_projects/index.html

Another option is to create a Python notebook that shows how to do lexmin optimization with CVXPY. We could include that here: https://www.cvxpy.org/examples/index.html

What do you think about those options?

**Reply by erelsgl** [reply]
I thought it could be more convenient to make it an objective type, so that it can be used like this:
 
     prob = cvxpy.Problem(cxvpy.Leximin([objective1, objective2]), constraints)

But the second option you mention (releasing this code as a small PyPi package) is reasonable too.

**Comment by erelsgl** [comment]
Finally, I managed to put my code into a library:

https://github.com/erelsgl/cvxpy_leximin

I would love comments about how to make this library more useful to `cvxpy` users.

**Reply by SteveDiamond** [reply]
It looks great!
